docs/advanced/atomic-swap.md

```diff
@@ -0,0 +1,33 @@
+# Atomic Swaps
+
+---
+
+## Overview
+
+An Atomic Swap is a smart contract technology which makes possible to exchange coins from two different blockchains without having to trust any third party, for example a centralized exchange.
+The smart contracts used in this case are not necessarily Turing-complete smart contracts (like in the system of Ethereum).
+Smart contracts also exist in the system of Bitcoin or Decred, or almost any other cryptocurrency.
+The scripting languages of these cryptocurrencies are not Turing-complete, which is intentionally designed like this to avoid loops, which would create security vulnerabilities.  
+For an atomic swap, a cryptocurrency must have the 'OP_CHECKLOCKTIMEVERIFY' opcode, which makes a timelock possible.
+The other prerequisite is that they need to have the same hash function.
+
+---
+
+## History of Atomic Swaps
+
+Tier Nolan is considered the inventor of atomic swaps. The concept of decentralized exchanges were widely discussed already in 2012. [The first atomic swap](https://blog.decred.org/2017/09/20/On-Chain-Atomic-Swaps/) happened between Decred and Litecoin on September 20, 2017.
+
+## Using Decred's atomicswap tool
+
+To perform an atomic swap, one needs to use the necessary command line tools for both chains. Each blockchain has it’s own tools. 
```

Comment:
Suggest separating the technical steps (using CLI tools) and the high-level steps into different paragraphs. 

This paragraph does a decent job laying out the high-level steps, showing you've done your research. However, I think it needs a little tightening up. As someone who knows a bit about blockchain, Decred, and smart contracts, there are some concepts that aren't clear to me just from reading this. I know this stuff is very hard to explain, and we don't have to define every concept here, but we should be as clear as possible. Below, I've started putting your steps into a list, with comments pointing out where I (the reader) am confused:

Below are the high-level steps for performing an atomic swap.
- The party initiating the swap creates a smart contract and a secret {SB: secret what? random number they create?}. A hash of the secret is visible to both parties.
- The receiving party audits the smart contract, verifying that the address {SB: which address? sender? receiver? both?}, amount, and timelock are correct. 
- If the audit is successful, the receiving party can participate in the contract.

Reply:
- Yep, the secret is essentially just a very very large number (32 bytes)
- The receiving party should validate that their own address is appearing correctly in the contract (ie. verify that they are actually going to receive the funds they are requesting)